qiskit/chemistry/results/electronic_structure_result.py

```diff
@@ -0,0 +1,346 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The electronic structure result."""
+
+from typing import List, Optional, Tuple, cast
+
+import logging
+import numpy as np
+
+from qiskit.aqua.algorithms import AlgorithmResult
+from qiskit.chemistry import QMolecule
+
+from .eigenstate_result import EigenstateResult
+
+logger = logging.getLogger(__name__)
+
+# A dipole moment, when present as X, Y and Z components will normally have float values for all
+# the components. However when using Z2Symmetries, if the dipole component operator does not
+# commute with the symmetry then no evaluation is done and None will be used as the 'value'
+# indicating no measurement of the observable took place
+DipoleTuple = Tuple[Optional[float], Optional[float], Optional[float]]
+
+
+class ElectronicStructureResult(EigenstateResult):
+    """The electronic structure result."""
+
+    @property
+    def algorithm_result(self) -> AlgorithmResult:
```

Comment:
Why do we need both algorithm_result here and raw_result in its parent class?

Reply:
I overlooked this field when copying from the `MolecularGroundStateResult`. I will remove it since we only want the `raw_result` field.